katsdpsigproc/accel.py

```diff
@@ -173,46 +173,15 @@ def all_devices() -> List[AbstractDevice]:
     return devices
 
 
-def create_some_context(
-        interactive: bool = True,
-        device_filter: Optional[Callable[[AbstractDevice], bool]] = None) -> AbstractContext:
-    """Create a single-device context, selecting a device automatically.
-
-    This is similar to `pyopencl.create_some_context`. A number of environment
-    variables can be set to limit the choice to a single device:
-
-     - KATSDPSIGPROC_DEVICE: device number from amongst all devices
-     - CUDA_DEVICE: CUDA device number (compatible with PyCUDA)
-     - PYOPENCL_CTX: OpenCL platform and optionally device number (compatible with PyOpenCL)
-
-    The first of these that is encountered takes effect. If it does not exist,
-    an exception is thrown.
-
-    Parameters
-    ----------
-    interactive
-        If true, and `sys.stdin.isatty()` is true, and there are multiple
-        choices, it will prompt the user. Otherwise, it will choose the first
-        available device, favouring CUDA over OpenCL, then GPU over
-        accelerators over other OpenCL devices.
-    device_filter
-        If specified, each device in turn is passed to it, and it must return
-        True to keep the device as a candidate or False to reject it.
-
-    Raises
-    ------
-    RuntimeError
-        If no device could be found or the user made an invalid selection
+def candidate_devices(device_filter: Optional[Callable[[AbstractDevice], bool]] = None) \
+        -> Sequence[AbstractDevice]:
```

Comment:
This was split out of `create_some_context` so that pytest_plugin can call it directly.